class.phpmailer.php

```diff
@@ -840,6 +840,14 @@ protected function addAnAddress($kind, $address, $name = '')
             }
             return false;
         }
+        if (function_exists('idn_to_ascii') && $this->hasMultiBytes($address)) { ##@TODO: Write our own "idn_to_ascii" function for PHP 5.2 and earlier.
+            // can't use $this->parseAddresses, because it calls validateAddress, which will fail here
+            $elements = explode('@',strrev($address),2);
```

Comment:
I rarely use strrpos(), but here IMHO it would make more readable and cheaper code than the triple strrev():

```php
if (false !== $pos = strrpos($address, '@')) {
    ++$pos;
    $address = substr($address, 0, $pos).idn_to_ascii(substr($address, $pos));
}
unset($pos);
```